internal/metrics/embedding_metrics.go

```diff
@@ -0,0 +1,112 @@
+// Copyright Envoy AI Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package metrics
+
+import (
+	"context"
+	"time"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+
+	"github.com/envoyproxy/ai-gateway/filterapi"
+	"github.com/envoyproxy/ai-gateway/filterapi/x"
+)
+
+const (
+	genaiOperationEmbedding = "embedding"
+)
+
+// embedding is the implementation for the embedding AI Gateway metrics.
+type embedding struct {
```

Comment:
Would it be possible generalize chatCompletion metric code? I think we can remove the entire copy-and-paste here